File: atcoder.jp/contests/abc202/tasks/abc202_b/main.hs
import qualified Data.Map as Map
import Data.Maybe (fromMaybe)

main :: IO ()
main = do
  s <- getLine

  let d = Map.fromList [('6', '9'), ('9', '6')]
      ans = map (translate d) $ reverse s

  putStrLn ans

translate :: Ord k => Map.Map k k -> k -> k
translate mp x = fromMaybe x $ Map.lookup x mp

File: atcoder.jp/contests/abc202/tasks/abc202_b/main.cpp
#include <bits/stdc++.h>
using namespace std;

string reverse(const string &s) { return {s.rbegin(), s.rend()}; }

string translate(const map<char, char> &mp, const string &s) {
  string ret;
  for (auto &&c : s) ret += mp.count(c) ? mp.at(c) : c;
  return ret;
}

int main() {
  string S;
  cin >> S;

  map<char, char> d = {{'6', '9'}, {'9', '6'}};
  string ans = translate(d, reverse(S));

  cout << ans << endl;
}
